### PR TITLE
Switch to GitHub actions for builds

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
Switches away from Travis and Appveyor to Github Actions for building and publishing.

Github Actions has several advantages:
 - It's built into Github directly - no need for a 3rd party service
 - It can build on all 3 required platforms: Unix, MacOS and Windows
 - A single workflow can have multiple runners and artifacts from each platform can be shared with other builds

I'll need to pair with someone that has admin access to this repo to setup the NPM_TOKEN secret for publishing. When this is ready to be merged, we can disconnect both Travis and Appveyor.

### CODE REVIEW

**Comment by brad4d** [comment]
@ChadKillingsworth, thanks for working on this.

I've added @nreid260 as a reviewer here, because he's actively working on streamlining our external release process.

Assuming you succeed in switching to GitHub Actions here, do you think it will make sense to do so for the primary google/closure-compiler repo as well? Does it fully replace the functionality we're currently getting from using TravisCI there?

File: build-scripts/build_js_compiler.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      })
+      .then(({exitCode}) => {
+        if (exitCode !== 0) {
+          process.exit(exitCode);
+          return;
+        }
+        // Add a license header to the gwt version jscomp.js file since the compiler build omits this.
+        // If the gwt version ever has a source map, the source mappings will need updated to account for the
+        // prepended lines.
+        const jscompFileContents = fs.readFileSync(compilerJsBinaryPath, 'utf8');
```
Comment (brad4d) [inline build-scripts/build_js_compiler.js:119]: Are you intending to land the PR for removing the JS-compiled version of closure-compiler before this PR?

File: build-scripts/graal-env.js [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 const GRAAL_OS = process.platform === 'win32' ? 'windows' : process.platform;
 const GRAAL_VERSION = process.env.GRAAL_VERSION || '20.1.0';
-const GRAAL_FOLDER = `graalvm-ce-java8-${GRAAL_OS}-amd64-${GRAAL_VERSION}`;
```
Comment (brad4d) [inline build-scripts/graal-env.js:20]: Is making this change dependent on also switching from JDK8 to JDK11? Could you make the JDK switch in a separate PR?

Comment (ChadKillingsworth) [reply build-scripts/graal-env.js:20]: Yes of course - part of this change was proving to myself that I could upgrade this - but it's trivial to just use Java 8 for now.

File: build-scripts/build_js_compiler.js (hunk 1 above)
Comment (ChadKillingsworth) [reply build-scripts/build_js_compiler.js:119]: I think I'd prefer to land this PR before removing the JS version. The number of different systems (Travis + Appveyeor + Github + NPM) and their interactions is incredibly complicated. It's much simplified after this.

Comment (ChadKillingsworth) [reply build-scripts/build_js_compiler.js:119]: Actually no I can't. Building the windows native image on Java 8 can only be done with a very very old windows SDK (windows 7). Appveyor is the only build system I could find to support that - and it's extremely difficult to maintain.

GitHub actions allowed updating to java 11 for the main build, but java 8 for GWT. This massively simplified the build.

File: build-scripts/graal-env.js (hunk 2 above)
Comment (ChadKillingsworth) [reply build-scripts/graal-env.js:20]: Actually no I can't. Building the windows native image on Java 8 can only be done with a very very old windows SDK (windows 7). Appveyor is the only build system I could find to support that - and it's extremely difficult to maintain.

GitHub actions allowed updating to java 11 for the main build, but java 8 for GWT. This massively simplified the build.

**Comment by ChadKillingsworth** [comment]
@brad4d I definitely think it's worth considering moving the main repo to github actions, but it's not near the problem the npm repo is. Namely the main repo only has to build on one OS.

File: build-scripts/npm-client.js [hunk 3]
```diff
@@ -80,27 +80,29 @@ switch (pkg.name) {
...
-            logToFile(logMessageQueue.join('\n'));
-          })
-          .catch(err => {
-            process.exitCode = 1;
-            logMessageQueue.push('   ❌ publish failed');
-            logToFile(logMessageQueue.join('\n'));
-            return Promise.reject(err);
-          });
+    // Github actions can publish all versions from the same run, but Travis and Appveyor can't
+    if (!process.env.GITHUB_ACTIONS) {
```
Comment (nreid260) [inline build-scripts/npm-client.js:84]: Is there still a world where Travis and Appveyor will be triggered?

Comment (ChadKillingsworth) [reply build-scripts/npm-client.js:84]: Only if the PR is merged before those 3rd party services are de-provisioned. I just wanted to be defensive here.

Comment (nreid260) [reply build-scripts/npm-client.js:84]: Ok, I'm not sure what the turn-around time on this is. Generally I thought it was low enough that we could squash this into a single PR, but you're better aware of the limitations here than I am.

Comment (ChadKillingsworth) [reply build-scripts/npm-client.js:84]: When we are ready to merge, I can disconnect those services, change that code block and do one more full run of the build to get all green.

Comment (ChadKillingsworth) [reply build-scripts/npm-client.js]: I removed this check and added scripts to ensure that any necessary artifacts exist in the package before publishing.
